### ISSUE TITLE
[Bug]: The signed URL in S3 is not rendering a file in a certain case.

### ISSUE THREAD

**Issue by AnandShahs123** [appsmithorg/appsmith#22272]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

If the file data type is "text" in S3 and the data format is base 64 in the filepicker widget, the signed URL is generated which doesn't rendered the file/ image.

Expected behaviour - Either the signed URL should not be generated OR the URL should render the result.

### Steps To Reproduce

1. Create a S3 datasource
2. Create a create a new file query
3. Use a file picker widget
4. set the data format as base64 in filepicker widget
5. set the data type as text in S3

Observe - A signed URL is generated which doesn't give the uploaded file

### Environment

Production

### Version

1.9.15

**Comment by piyushkrmaurya** [comment]
Tried to repro. If I understand correctly, you selected "Text/Binary" as "File Data Type" in the query.
This looks fine to me. The widget uploads the file in base64 format. The uploaded file gets converted to base64 encoding. The same base64 encoded data gets saved in text format when you upload the file. You should select "base64" in the query as well, so that it renders the encoding correctly.

**Comment by AnandShahs123** [comment]
Yes, I understand ideally a developer should select "base64" in the query. However, if by mistake he doesn't do so .. we should show an error. @piyushkrmaurya
